hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -399,19 +400,22 @@ private void handleUserTransaction(
                     networkUtilizationManager.trackFeePayments(payer, consensusNow, stack);
                 }
                 recordBuilder.status(validationResult.responseCodeEnum());
-
                 try {
-                    if (validationResult.status() == NODE_DUE_DILIGENCE_FAILURE) {
-                        feeAccumulator.chargeNetworkFee(creator.accountId(), fees.networkFee());
-                    } else if (validationResult.status() == PAYER_UNWILLING_OR_UNABLE_TO_PAY_SERVICE_FEE) {
-                        // We do not charge partial service fees; if the payer is unwilling or unable to cover
-                        // the entire service fee, then we only charge network and node fees (prioritizing
-                        // the network fee in case of a very low payer balance)
-                        feeAccumulator.chargeFees(payer, creator.accountId(), fees.withoutServiceComponent());
-                    } else {
-                        feeAccumulator.chargeFees(payer, creator.accountId(), fees);
+                    // If the payer is authorized to waive fees, then we don't charge them
+                    if (!hasWaivedFees) {
+                        if (validationResult.status() == NODE_DUE_DILIGENCE_FAILURE) {
+                            feeAccumulator.chargeNetworkFee(creator.accountId(), fees.networkFee());
+                        } else if (validationResult.status() == PAYER_UNWILLING_OR_UNABLE_TO_PAY_SERVICE_FEE) {
+                            // We do not charge partial service fees; if the payer is unwilling or unable to cover
+                            // the entire service fee, then we only charge network and node fees (prioritizing
+                            // the network fee in case of a very low payer balance)
+                            feeAccumulator.chargeFees(payer, creator.accountId(), fees.withoutServiceComponent());
+                        } else {
+                            feeAccumulator.chargeFees(payer, creator.accountId(), fees);
+                        }
                     }
                 } catch (final HandleException ex) {
+                    ex.printStackTrace();
```

Comment:
`log.error()` instead? Curious why the use of `printStackTrace()`?

Reply:
No it was all debugging. Removed it. Thanks